File: fibonacci.ts
/**
 * Calculates the Fibonacci number at a given position
 * @param n - The position in the Fibonacci sequence (0-indexed)
 * @returns The Fibonacci number at position n
 */
export function fibonacci(n: number): number {
  if (n < 0) {
    throw new Error("Fibonacci index cannot be negative");
  }

  if (n <= 1) {
    return n;
  }

  let prev = 0;
  let current = 1;

  for (let i = 2; i <= n; i++) {
    const next = prev + current;
    prev = current;
    current = next;
  }

  return current;
}

/**
 * Calculates the first n Fibonacci numbers
 * @param count - The number of Fibonacci numbers to generate
 * @returns An array containing the first n Fibonacci numbers
 */
export function fibonacciSequence(count: number): number[] {
  if (count < 0) {
    throw new Error("Count cannot be negative");
  }

  if (count === 0) {
    return [];
  }

  const sequence: number[] = [0];

  if (count === 1) {
    return sequence;
  }

  sequence.push(1);

  for (let i = 2; i < count; i++) {
    sequence.push(sequence[i - 1] + sequence[i - 2]);
  }

  return sequence;
}

// Example usage
console.log(`Fibonacci(25) = ${fibonacci(25)}`);
console.log(`First 10 Fibonacci numbers: ${fibonacciSequence(10).join(", ")}`);

File: fibonacci.py
"""Fibonacci number calculator module."""


def fibonacci(n: int) -> int:
    """
    Calculate the nth Fibonacci number.
    
    The Fibonacci sequence starts with 0 and 1, and each subsequent number
    is the sum of the previous two numbers.
    
    Args:
        n: The position in the Fibonacci sequence (0-indexed)
    
    Returns:
        The nth Fibonacci number
    
    Raises:
        ValueError: If n is negative
    """
    if n < 0:
        raise ValueError("n must be non-negative")
    
    if n <= 1:
        return n
    
    a, b = 0, 1
    for _ in range(2, n + 1):
        a, b = b, a + b
    
    return b


def fibonacci_sequence(n: int) -> list[int]:
    """
    Generate the first n Fibonacci numbers.
    
    Args:
        n: The number of Fibonacci numbers to generate
    
    Returns:
        A list containing the first n Fibonacci numbers
    
    Raises:
        ValueError: If n is negative
    """
    if n < 0:
        raise ValueError("n must be non-negative")
    
    sequence = []
    for i in range(n):
        sequence.append(fibonacci(i))
    
    return sequence
